### PR TITLE
Add Rundeck exporter

### PR DESCRIPTION
**Pull request by phsmith**

### CODE REVIEW

File: content/docs/instrumenting/exporters.md [hunk 1]
```diff
@@ -140,6 +140,7 @@ wide variety of JVM-based applications, for example [Kafka](http://kafka.apache.
    * [Rancher exporter](https://github.com/infinityworksltd/prometheus-rancher-exporter)
    * [Speedtest exporter](https://github.com/nlamirault/speedtest_exporter)
    * [Tankerkönig API Exporter](https://github.com/lukasmalkmus/tankerkoenig_exporter)
+   * [Rundeck exporter](https://github.com/phsmith/rundeck_exporter)
```
Comment (brian-brazil) [inline content/docs/instrumenting/exporters.md:143]: Keep it sorted.

Comment (brian-brazil) [inline content/docs/instrumenting/exporters.md:143]: This is not a SaaS API, so belongs down in misc.

**Comment by phsmith** [comment]
@brian-brazil Thanks for all instructions! 
I'll work to make the exporter better.

**Comment by phsmith** [comment]
@brian-brazil, please, could you check the new code to see if I'm on the right path: https://github.com/phsmith/rundeck_exporter/tree/feature/v2

**Comment by brian-brazil** [comment]
That looks better, the project metric still needs improvement to be useful.

**Comment by phsmith** [comment]
> That looks better, the project metric still needs improvement to be useful.

@brian-brazil I've made some modifications on the project metrics, thing that's better now.
https://github.com/phsmith/rundeck_exporter/tree/feature/v2

**Comment by brian-brazil** [comment]
The status should be a separate gauge, and you can also have the various times as gauges.

As a minor style thing I'd do executions -> execution too.

**Comment by phsmith** [comment]
> The status should be a separate gauge, and you can also have the various times as gauges.

@brian-brazil  You meant something like that:
`
rundeck_project_job_test_execution_info{execution_id="1623097",job_id="job-test",job_name="Job Test"} 1.0
rundeck_project_job_test_execution_status_info{execution_id="1623097",job_id="job-test",job_name="Job Test"} 1.0
`
And about the Rundeck meters and timers, I'm getting only the count metric, so I thing that it represents a counter.

**Comment by brian-brazil** [comment]
No,

`rundeck_project_execution_status{execution_id="1623097",job_id="job-test",job_name="Job Test"} 0`

for example.

> And about the Rundeck meters and timers, I'm getting only the count metric, so I thing that it represents a counter.

Yes, those are counters - and if a new enough version of Dropwizard is being used the timers can be made into summaries.

**Comment by phsmith** [comment]
@brian-brazil what about that:

```
rundeck_project_aic_helloworld_ansible_playbook_execution_status{job_id="helloworld-ansible-playbook-job-test",job_name="Job Test"} 2.0
rundeck_project_bdh_oracle_install_client_execution_status{job_id="dfd6d2aa-5159-46db-a0c5-0b9a8aa04527",job_name="Client Oracle Install"} 0.0
rundeck_project_tbs_servcom_install_execution_status{job_id="servcom_install",job_name="Servcom Client"} 0.0
```

**Comment by brian-brazil** [comment]
Bools like this should only be 0/1, and you have a label value in a metric name there which makes life harder for users.

**Comment by phsmith** [comment]
> Bools like this should only be 0/1, and you have a label value in a metric name there which makes life harder for users.

Projects executions on Rundeck have 4 status: succeeded, running, failed or aborted, that's the reason that I've put that on a Info Metric before. 
And the job_id can be a uuid or a name.
So, what should I do with that metric?

**Comment by brian-brazil** [comment]
Hmm, I'd presumed that there were only the completed ones. One approach is to have a gauge with a series for each state, and a value of 1 for the one that is true and 0 for the others.

The other approach is to report the status of the most recent non-running job and whether it succeeded or not as one bool.

**Comment by phsmith** [comment]
> Hmm, I'd presumed that there were only the completed ones. One approach is to have a gauge with a series for each state, and a value of 1 for the one that is true and 0 for the others.
> 
> The other approach is to report the status of the most recent non-running job and whether it succeeded or not as one bool.

@brian-brazil 
So, this is a good approach?

```
rundeck_project_tbs_servcom_install_execution_succeeded{job_id="servcom_install",job_name="Servcom Client Install"} 1.0
rundeck_project_tbs_servcom_install_execution_running{job_id="servcom_install",job_name="Servcom Client Install"} 0.0
rundeck_project_tbs_servcom_install_execution_failed{job_id="servcom_install",job_name="Servcom Client Install"} 0.0
rundeck_project_tbs_servcom_install_execution_aborted{job_id="servcom_install",job_name="Servcom Client Install"} 0.0
rundeck_project_tbs_servcom_install_execution_unknown{job_id="servcom_install",job_name="Servcom Client Install"} 0.0
```

**Comment by brian-brazil** [comment]
That's not the best approach. The status should be as a label, plus you still have the job name in the metric name.

**Comment by phsmith** [comment]
> That's not the best approach. The status should be as a label, plus you still have the job name in the metric name.

I've made in that format based on what you've mention before, I presume that I understand that wrong:

> Hmm, I'd presumed that there were only the completed ones. One approach is to have a gauge with a series for each state, and a value of 1 for the one that is true and 0 for the others.

So I have to add the project and the job name as a metric name? Like that?

```
rundeck_project_tbs_servcom_install_Servcom_Client_Install_execution{status="succeeded"} 1.0
rundeck_project_tbs_servcom_install_Servcom_Client_Install_execution{status="failed"} 0.0
```

Or like that is better:
```
rundeck_project_tbs_servcom_install_execution_status{job_id="servcom_install",job_name="Servcom Client Install",status="suceeded"} 1.0
rundeck_project_tbs_servcom_install_execution_status{job_id="servcom_install",job_name="Servcom Client Install",status="failed"} 0.0
```

Sorry for the newbie questions, if you known about some exporter that have similar metrics exposition, I'll be pleased to check  it.

**Comment by brian-brazil** [comment]
More like:

```
rundeck_project_execution_status{job_id="servcom_install",job_name="Servcom Client Install",status="succeeded"} 1.0
rundeck_project_execution_status{job_id="servcom_install",job_name="Servcom Client Install",status="failed"} 0.0
rundeck_project_execution_status{job_id="servcom_install",job_name="Servcom Client Install",status="aborted"} 0.0
```

> if you known about some exporter that have similar metrics exposition, I'll be pleased to check it.

Kube-state-metrics is where this particular pattern first came up.

**Comment by phsmith** [comment]
@brian-brazil Could check again how it is now please:
https://github.com/phsmith/rundeck_exporter/tree/feature/v2

**Comment by brian-brazil** [comment]
That metric looks fine.

**Comment by phsmith** [comment]
> That metric looks fine.

@brian-brazil So do you think that I can publish this version?

**Comment by brian-brazil** [comment]
I do, and then this can be merged.

**Comment by phsmith** [comment]
> I do, and then this can be merged.

Nice @brian-brazil! 
New version published.
Thanks a lot for the great support.

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**